runtime/gc_vlhgc/HeapRegionDataForAllocate.cpp

```diff
@@ -180,15 +184,23 @@ MM_HeapRegionDataForAllocate::taskAsArrayletLeaf(MM_EnvironmentBase *env)
  * or assigned to a new list immediately.
  */
 void 
-MM_HeapRegionDataForAllocate::removeFromArrayletLeafList()
+MM_HeapRegionDataForAllocate::removeFromArrayletLeafList(MM_EnvironmentVLHGC *env)
 {
 	Assert_MM_true(_region->isArrayletLeaf());
 	
 	MM_HeapRegionDescriptorVLHGC *next = _nextArrayletLeafRegion;
 	MM_HeapRegionDescriptorVLHGC *previous = _previousArrayletLeafRegion;
 	
 	Assert_MM_true(NULL != previous);
-	
+
+	/**
+	 * Restore/Recommit arraylet leaves that have been previously decommitted.
+	 */
+	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(env);
```

Comment:
this should probably be under 'if offheap enabled'

Reply:
this does not seem to be resolved (again, just the new part, committing shoudl be conditional for offheap enabled)